source/mainwindow.cpp

```diff
@@ -542,7 +549,38 @@ void MainWindow::saveFile(SaveAsParam *params)
     this->ui->statusBar->showMessage("Saving...");
     this->ui->statusBar->repaint();
 
-    bool change = this->cel->save(params);
+    bool change = false;
+    QString filePath;
+    if (params != nullptr)
+        filePath = params->celFilePath;
+    if (filePath.isEmpty())
+        filePath = this->gfx->getFilePath();
+    if (this->gfx->getType() == D1CEL_TYPE::V1_LEVEL) {
+        if (!filePath.toLower().endsWith("cel")) {
+            QMessageBox::StandardButton reply;
+            reply = QMessageBox::question(nullptr, "Confirmation", "Are you sure you want to save as " + filePath + "? Data conversion is not supported.", QMessageBox::Yes | QMessageBox::No);
+            if (reply != QMessageBox::Yes) {
+                // Clear loading message from status bar
+                this->ui->statusBar->clearMessage();
+                return;
+            }
+        }
+        change = D1CelTileset::save(*this->gfx, params);
+    } else {
+        if (filePath.toLower().endsWith("cel")) {
+            change = D1Cel::save(*this->gfx, params);
+        } else if (filePath.toLower().endsWith("cl2")) {
+            change = D1Cl2::save(*this->gfx, false, params);
+        } else if (filePath.toLower().endsWith("clx")) {
+            change = D1Cl2::save(*this->gfx, true, params);
+        } else {
+            QMessageBox::critical(nullptr, "Warning", "Not supported.");
```

Comment:
This should probably be an `Error` instead of `Warning` since we where unable to comply with the request.